src/application/components/Explorer/Explorer.tsx

```diff
@@ -104,42 +116,88 @@ function executeOperation({
-        const checkForSubscriptionTermination = (event: MessageEvent) => {
-          if(event.data.name.startsWith(EXPLORER_SUBSCRIPTION_TERMINATION)) {
-            sendSubscriptionTerminationRequest();
-            observer.complete();
-            window.removeEventListener('message', checkForSubscriptionTermination);
-          }
-        }
+    listenForResponse(
+      (response) => {
+        observer.next(response);
+        if (isSubscription) {
+          const checkForSubscriptionTermination = (event: MessageEvent) => {
+            if (event.data.name.startsWith(EXPLORER_SUBSCRIPTION_TERMINATION)) {
+              sendSubscriptionTerminationRequest();
+              observer.complete();
+              window.removeEventListener(
+                "message",
+                checkForSubscriptionTermination
+              );
+            }
+          };
 
-        window.addEventListener('message', checkForSubscriptionTermination);
-      }
-      else {
-        observer.complete();
-      }
-    }, operationName, !!isSubscription,);
+          window.addEventListener("message", checkForSubscriptionTermination);
+        } else {
+          observer.complete();
+        }
+      },
+      operationName,
+      !!isSubscription
+    );
   });
 }
 
-export const Explorer = ({ navigationProps, embeddedExplorerProps }: {
+const getGraphRefFromLocalStorage = () => {
+  const graphRefsByParentHostnameString =
+    window.localStorage.getItem("explorerGraphRefs");
+  const graphRefsByParentHostname = graphRefsByParentHostnameString
+    ? JSON.parse(graphRefsByParentHostnameString)
+    : {};
+  return graphRefsByParentHostname["temp-key"];
+};
+
+const setGraphRefFromLocalStorage = (graphRef: string) => {
+  const graphRefsByParentHostnameString =
+    window.localStorage.getItem("explorerGraphRefs");
+  const graphRefsByParentHostname = graphRefsByParentHostnameString
+    ? JSON.parse(graphRefsByParentHostnameString)
+    : {};
+  graphRefsByParentHostname["temp-key"] = graphRef;
+  window.localStorage.setItem(
+    "explorerGraphRefs",
+    JSON.stringify(graphRefsByParentHostname)
+  );
+};
+
+export const Explorer = ({
+  isVisible,
+  navigationProps,
+  embeddedExplorerProps,
+}: {
+  isVisible: boolean | undefined;
   navigationProps: {
-    queriesCount: number,
-    mutationsCount: number,
-  }
+    queriesCount: number;
+    mutationsCount: number;
+  };
   embeddedExplorerProps: {
-    embeddedExplorerIFrame: HTMLIFrameElement | null,
-    setEmbeddedExplorerIFrame: (iframe: HTMLIFrameElement) => void,
-  }
+    embeddedExplorerIFrame: HTMLIFrameElement | null;
+    setEmbeddedExplorerIFrame: (iframe: HTMLIFrameElement) => void;
+  };
 }): jsx.JSX.Element => {
-  const [schema, setSchema] = useState<IntrospectionQuery | null>(null)
+  const [graphRef, setGraphRef] = useState<string>();
+  const [showGraphRefModal, setShowGraphRefModal] = useState<
+    false | "triggeredByIntrospection" | "triggeredManually"
```

Comment:
minor nit, could we rename this to `triggeredByIntrospectionFailure` 🙏 was initially confused about why we'd want to show the modal when we receive a successful introspection